internal/handlers/common/update_array_operators.go

```diff
@@ -91,6 +97,40 @@ func processPopArrayUpdateExpression(doc *types.Document, update *types.Document
 	return changed, nil
 }
 
+// ensureNonExistentPrefix checks if a given path and its prefix
+// does not exist in doc. It returns error if any prefix exists in the path.
```

Comment:
Let's specify what type of error it returns as it looks important (that's the error we want to return in the response).

Reply:
Thanks I updated the comment